repository/src/main/java/org/alfresco/repo/action/executer/ActionExecuterAbstractBase.java

```diff
@@ -79,10 +85,14 @@ public abstract class ActionExecuterAbstractBase extends ParameterizedItemAbstra
      */
     public void init()
     {
-        if (this.publicAction == true)
+        if (this.publicAction)
         {
             this.runtimeActionService.registerActionExecuter(this);
         }
+
+        if (!this.actionAccessRestrictions.isEmpty()) {

```

Comment:
Does it make sense to replace this registration by obtaining an executer bean from application context in the `ActionServiceImpl` ?

Reply:
Changed, we'll end up with Map being filled as requests are made to different ActionExecuter instances as opposed to only access restricted ones, but they aren't extensive enough to pose an issue.